app/assets/javascripts/discourse/app/lib/textarea-text-manipulation.js

```diff
@@ -64,6 +65,11 @@ export default class TextareaTextManipulation {
     return this.textarea.value;
   }
 
+  set value(value) {
+    this.textarea.value = value;
+    this.textarea.dispatchEvent(new Event("change", { bubbles: true }));
+  }
+
```

Comment:
Im ok with keeping it like this but Im mixed on the API being:

```
textareaManipulation.value
textareaManipulation.value =
```

While it might seem nice, I don't expect to be able to set the value on something called "textareaManipulation". And maybe going with `getValue` and `setValue` would actually be a better API here.

Reply:
Yeah, I agree it would be nice to make it a bit more explicit. Especially since `textarea.value = ` causes the undo/redo stack to be thrown away! This isn't something we want to be doing often.

Reply:
The only way to avoid throwing away the undo stuff is with the surgical `execCommand` technique we use.

Setting value directly is fine in some cases (e.g. when initially rendering). But we should avoid doing it in response to buttons or keyboard-shortcuts.

Reply:
I've removed this use of the setter as well as two others from https://github.com/discourse/discourse/pull/29973, `applyList` and `formatCode`.

These actions shouldn't reset the undo history anymore.